src/AmplNLWriter.jl

```diff
@@ -135,24 +135,70 @@ type AmplNLMathProgModel <: AbstractMathProgModel
             "")
     end
 end
+type AmplNLLinearQuadraticModel <: AbstractLinearQuadraticModel
+    inner::AmplNLMathProgModel
+end
+type AmplNLNonlinearModel <: AbstractNonlinearModel
+    inner::AmplNLMathProgModel
+end
+
+# Macro to generate wrappers for SolverInterface functions
+# e.g.
+#   @wrap status(m::AmplNLMathProgModel) = m.status
+# produces
+#   status(m::AmplNLMathProgModel) = m.status
+#   status(outer::AmplNLNonlinearModel) = status(outer.inner)
+#   status(outer::AmplNLLinearQuadraticModel) = status(outer.inner)
+macro wrap(args)
```

Comment:
It's a bit of a pain for the ones with extra parameters though, unless there's a smart way?